osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs

```diff
@@ -187,28 +187,18 @@ private bool snapToVisibleBlueprints(Vector2 screenSpacePosition, out SnapResult
                 if (b.IsSelected)
                     continue;
 
-                var hitObject = (OsuHitObject)b.Item;
+                var snapPositions = b.ScreenSpaceSnapPoints;
 
-                Vector2? snap = checkSnap(hitObject.Position);
-                if (snap == null && hitObject.Position != hitObject.EndPosition)
-                    snap = checkSnap(hitObject.EndPosition);
+                var filteredSnapPositions = snapPositions.Cast<Vector2?>().FirstOrDefault(p => Vector2.Distance(p.Value, screenSpacePosition) < snapRadius);
```

Comment:
Aside from the code inspection, my primary question here would be - if there are potentially multiple snap points matching within the snap radius, then why would the first one be chosen arbitrarily, rather than the closest one? (i.e. `.MinBy(distance)`).

Reply:
The `ScreenSpaceSelectionPoint` is always first in `ScreenSpaceSnapPoints`, so it priorities this `SelectionPoint` (basicly sliderhead first, then sliderend like it worked before).

I just checked differences between taking first in snap range and closest one. If sliderhead and sliderend are almost perfectly at the top of each other, then the latter solution allowed a bit easier switching between points.

If the edge case is whatsoever significant then sure I can change it to the one you suggested.

Reply:
I don't map so I couldn't really say on that end, but it feels like a more natural way of defining this operation is all...

Reply:
I think I'd prefer the proposed solution.

Please also note the CI code inspection failure on this line.